static/src/javascripts/projects/common/modules/identity/upsell/account-creation/AccountCreationFlow.js

```diff
@@ -0,0 +1,209 @@
+            isError: false,
+        });
+
+        reqwest({
+            url: '/password/guest',
+            method: 'post',
+            data: {
+                csrfToken: this.props.csrfToken,
+                token: this.props.accountToken,
+                password: this.state.password,
+            },
+            success: () => {
+                this.props.onAccountCreated();
+            },
+            error: response => {
+                try {
+                    const apiError = JSON.parse(response.responseText)[0];
+                    this.setState({
+                        isError: true,
+                        errorReason: apiError.description,
+                    });
+                } catch (exception) {
+                    this.setState({ isError: true });
+                }
+            },
+            complete: () => {
+                this.setState({ isLoading: false });
+            },
+        });
+    };
+
+    handlePasswordChange = (ev: Event) => {
+        if (!(ev.target instanceof HTMLInputElement)) {
+            return;
+        }
+        this.setState({ password: ev.target.value });
+    };
+
+    render() {
+        const { isError, errorReason } = this.state;
+        return (
+            <form className="form" onSubmit={this.onSubmit}>
+                <hr className="manage-account-small-divider" />
+                {isError && (
+                    <div className="form__error">
+                        {errorReason || 'Oops. Something went wrong'}
+                    </div>
+                )}
+                <h1 className="identity-title--small">
+                    Want even more from the Guardian? Create a free account
+                </h1>
+                <div className="identity-forms-message__body">
+                    <p>
+                        Create your account by setting a password, and manage
+                        your email preferences at any time.
+                    </p>
+                </div>
+                <div className="fieldset__fields">
+                    <li className="form-field" id="password_field">
+                        <label htmlFor="password">
+                            <div className="label">Password</div>
+                            <div className="input">
+                                <input
+                                    className="text-input"
+                                    type="password"
+                                    id="password"
+                                    name="password"
+                                    value={this.state.password}
+                                    autoComplete="off"
+                                    onChange={this.handlePasswordChange}
+                                    autoCapitalize="off"
+                                    autoCorrect="off"
+                                    spellCheck="false"
+                                    aria-required="true"
```

Comment:
It looks like this input field should also have the attribute `required`? There's no check in `onSubmit` that this input (`this.state.password`) has any value?

Reply:
Yep it should definitely be required, there's server side validation too so we show an error if it was a bad value or empty.